lib/flutter/flutter_sdk.dart

```diff
@@ -232,23 +232,12 @@ class FlutterTool {
 
   FlutterTool(this.sdk, this.toolPath);
 
-  ProcessRunner runRaw(List<String> args, {
-    String cwd,
-    bool startProcess: true
-  }) {
-    // Run process under bash on the mac, to capture the user's env variables.
-    if (isMac) {
-      //exec('/bin/bash', ['-l', '-c', 'which dart'])
-      String arg = toolPath + ' ' + args.join(' ');
-      ProcessRunner runner = new ProcessRunner(
-          '/bin/bash', args: ['-l', '-c', arg], cwd: cwd);
-      if (startProcess) runner.execStreaming();
-      return runner;
-    } else {
-      ProcessRunner runner = new ProcessRunner(toolPath, args: args, cwd: cwd);
-      if (startProcess) runner.execStreaming();
-      return runner;
-    }
+  ProcessRunner runRaw(List<String> args,
+      {String cwd, bool startProcess: true}) {
+    ProcessRunner runner =
+        new ProcessRunner.underShell(toolPath, args: args, cwd: cwd);
```

Comment:
We should still just run `ProcessRunner.underShell()` here on the mac. We don't have the shell issues on linux, and it's likely that `underShell()` will not work on windows (or always emit the warning). You could also move the system check into `underShell()` itself.